jetpack/chart/utils.py

```diff
@@ -86,26 +83,89 @@ def noticks(**kwargs):
 
 
 @axwrapper
-def nospines(**kwargs):
+def nospines(left=False, bottom=False, top=True, right=True, **kwargs):
     """
-    Hides the top and rightmost axis spines
+    Hides the specified axis spines (by default, right and top spines)
     """
 
     ax = kwargs['ax']
 
-    # disable spines
-    ax.spines['right'].set_color('none')
-    ax.spines['top'].set_color('none')
+    # assemble args into dict
+    disabled = dict(left=left, right=right, top=top, bottom=bottom)
 
-    # disable ticks
-    ax.yaxis.set_ticks_position('left')
-    ax.xaxis.set_ticks_position('bottom')
+    # disable spines
+    for key in disabled:
+        if disabled[key]:
+            ax.spines[key].set_color('none')
+
+    # disable xticks
+    if disabled['top'] and disabled['bottom']:
+        ax.set_xticks([])
+    elif disabled['top']:
+        ax.xaxis.set_ticks_position('bottom')
+    elif disabled['bottom']:
+        ax.xaxis.set_ticks_position('top')
+
+    # disable yticks
+    if disabled['left'] and disabled['right']:
+        ax.set_yticks([])
+    elif disabled['left']:
+        ax.yaxis.set_ticks_position('right')
+    elif disabled['right']:
+        ax.yaxis.set_ticks_position('left')
 
     return ax
 
+@axwrapper
+def minlabels(x=True, y=True, n_xticks=4, n_yticks=4, **kwargs):
+    """
+    Label only the first and last tick marks.
+    """
+    ax = kwargs['ax']
+
+    if x:
+        # get first and last tick
+        xmin, xmax = ax.get_xlim()
+
+        # remove decimals from labels
+        if xmin.is_integer():
+            xmin = int(xmin)
+        if xmax.is_integer():
+            xmax = int(xmax)
+
+        # reset tick marks
+        xt = np.linspace(xmin, xmax, n_xticks)
+
+        # update plot
+        xlab = [str(xmin), *['' for _ in range(len(xt)-2)], str(xmax)]
```

Comment:
This line is causing errors in python 3.3 and 3.4, can you create this another way?